### PR TITLE
feat(perfmon): Add native performance monitoring docs

### PR DESCRIPTION
**Pull request by relaxolotl**
A lot of this has been blatantly copied from https://github.com/getsentry/sentry-docs/pull/4610's changes, and updated appropriately for native. The contents of the prose-heavy docs in this significantly differ from the rust PR, however.

Credit for a lot of this goes to @loewenheim.

### CODE REVIEW

File: src/includes/performance/distributed-tracing/native.mdx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+int main(int argc, char **argv) {
+  ...
+	// transaction to continue off of
+	sentry_transaction_context_t *tx_ctx
+        = sentry_transaction_context_new("honk", NULL);
+    sentry_transaction_t *tx = sentry_transaction_start(tx_ctx);
+
+	sentry_value_t *headers = sentry_value_new_object();
+	sentry_transaction_iter_headers(tx, copy_headers_to, (void *)headers);
+}
```
Comment (relaxolotl) [inline src/includes/performance/distributed-tracing/native.mdx:33]: @native, probably most importantly @Swatinem: i'll probably need a sanity check on the code and the example on this file.

**Comment by relaxolotl** [comment]
putting this in draft until i can figure out how to fix hyperlink's complaints about various native backend pages linking to nonexistent sampling pages

**Comment by loewenheim** [comment]
For context, the links it complains about are
```
  platforms/native/guides/breakpad/configuration/sampling
  platforms/native/guides/crashpad/configuration/sampling
  platforms/native/guides/minidumps/configuration/sampling
  platforms/native/guides/ue4/configuration/sampling
  platforms/native/guides/wasm/configuration/sampling
```

**Review by imatwawana — COMMENTED** [review]
I know this is still in draft, but I wanted to make some suggestions for simplifying the language and avoiding passive voice. Otherwise though, this looks great!

**Comment by relaxolotl** [comment]
fixed the links

**Comment by relaxolotl** [comment]
@imatwawana thanks for the early review; I've addressed the feedback you've left behind while fixing up this branch.

I've also added in additional text during the process of repairing and getting this branch to build, which I'd appreciate your feedback on as well. They're mixed into what you've already reviewed, so here's a list of the files that could use a second look:
- src/includes/performance/always-inherit-sampling-decision/native.mdx
- src/includes/performance/enable-tracing/native.mdx
- src/includes/performance/retrieve-transaction/native.mdx
- src/includes/performance/uniform-sample-rate/native.mdx (the last ~3 lines)

Other than the last item, everything listed above are new files with a very small amount of content (1-2 sentences).

**Review by loewenheim — APPROVED** [review]

File: src/includes/performance/always-inherit-sampling-decision/native.mdx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<Alert level="warning" title="Sampling Functions">
+The Native SDK currently does not support sampling functions (<PlatformIdentifier name="traces-sampler" />).
```
Comment (imatwawana) [inline src/includes/performance/always-inherit-sampling-decision/native.mdx:2]:
```suggestion
<Alert level="warning" title="Sampling Functions">

The Native SDK currently does not support sampling functions (<PlatformIdentifier name="traces-sampler" />).

```

File: src/includes/performance/uniform-sample-rate/native.mdx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+To do this, set the <PlatformIdentifier name="traces-sample-rate" /> option to a number between 0 and 1. With this option set, every transaction created will have that percentage chance of being sent to Sentry. For example, if you set <PlatformIdentifier name="traces-sample-rate" /> to `0.2`, approximately 20% of your transactions will be recorded and sent. That looks like this:
+
+<PlatformContent includePath="performance/traces-sample-rate" />
+
+<Alert level="warning" title="Sampling Functions">
+The Native SDK currently does not support setting a sampling function (<PlatformIdentifier name="traces-sampler" />).
```
Comment (imatwawana) [inline src/includes/performance/uniform-sample-rate/native.mdx:6]:
```suggestion
<Alert level="warning" title="Sampling Functions">

The Native SDK currently does not support setting a sampling function (<PlatformIdentifier name="traces-sampler" />).

```

**Review by imatwawana — APPROVED** [review]
A couple small style changes, but otherwise, looks good!

**Pull request merged**